recipes/quill/all/conanfile.py

```diff
@@ -0,0 +1,100 @@
+from conans import ConanFile, CMake, tools
+from conans.model.version import Version
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class QuillConan(ConanFile):
+    name = "quill"
+    description = "C++14 Asynchronous Low Latency Logging Library"
+    homepage = "https://github.com/odygrd/quill/"
+    topics = ("conan", "quill", "logging", "log")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"fmt": [True, False]}
+    default_options = {"fmt": False}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        supported_archs = ["x86", "x86_64"]
+
+        if not any(arch in str(self.settings.arch) for arch in supported_archs):
+            raise ConanInvalidConfiguration("{} is not supported by {}".format(self.settings.arch, self.name))
+
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "14")
+
+        minimum_version = {
+            "clang": "5",
+            "gcc": "5",
+            "Visual Studio": "15.0",
+        }.get(str(self.settings.compiler))
+
+        if not minimum_version:
+            self.output.warn(
+                "Unknown compiler {} {}. Assuming compiler supports C++14".format(self.settings.compiler,
+                                                                                  self.settings.compiler.version))
+
+        else:
+            version = Version(self.settings.compiler.version.value)
+            if version < minimum_version:
+                raise ConanInvalidConfiguration(
+                    "The compiler {} {} does not support C++14".format(self.settings.compiler,
+                                                                       self.settings.compiler.version))
+
+    def requirements(self):
+        if self.options.fmt:
+            self.requires("fmt/6.1.2")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["QUILL_FMT_EXTERNAL"] = self.options.fmt
+        self._cmake.configure()
+        
+        return self._cmake
+
+    def build(self):
+        if "patches" in self.conan_data and self.version in self.conan_data["patches"]:
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch)
+
+        cmake = self._configure_cmake()
```

Comment:
```suggestion
        cmake = self._configure_cmake(build_folder=self._build_subfolder)
```

Reply:
thanks, I saw your message after I had already pushed and I did something similar just configuring it twice. 
It still fails to link the `test_package` now like a few days ago.
It doesn't fail when I try locally with 
`C:\dev\quill\all>conan test test_package quill/1.2.1@ody/test`